src/main/java/io/cryostat/net/web/WebServer.java

```diff
@@ -203,21 +208,56 @@ public void start() throws FlightRecorderException, SocketException, UnknownHost
         this.server.requestHandler(
                 req -> {
                     Instant start = Instant.now();
+                    WebServerRequest evt =
+                            new WebServerRequest(
+                                    req.remoteAddress().toString(),
+                                    req.method().toString(),
+                                    req.path());
+                    evt.begin();
+
                     req.response()
                             .endHandler(
-                                    (res) ->
-                                            logger.info(
-                                                    "({}): {} {} {} {}ms",
-                                                    req.remoteAddress().toString(),
-                                                    req.method().toString(),
-                                                    req.path(),
-                                                    req.response().getStatusCode(),
-                                                    Duration.between(start, Instant.now())
-                                                            .toMillis()));
+                                    (res) -> {
+                                        logger.info(
+                                                "({}): {} {} {} {}ms",
+                                                req.remoteAddress().toString(),
+                                                req.method().toString(),
+                                                req.path(),
+                                                req.response().getStatusCode(),
+                                                Duration.between(start, Instant.now()).toMillis());
+                                        evt.setStatusCode(req.response().getStatusCode());
+                                        evt.end();
+                                        if (evt.shouldCommit()) {
+                                            evt.commit();
+                                        }
+                                    });
                     router.handle(req);
                 });
     }
 
+    @Name("io.cryostat.net.web.WebServer.WebServerRequest")
+    @Label("Web Server Request")
+    @Category("Cryostat")
+    @SuppressFBWarnings(
+            value = "URF_UNREAD_FIELD",
+            justification = "The event fields are recorded with JFR instead of accessed directly")
+    public static class WebServerRequest extends Event {
+        String remoteAddr;
```

Comment:
In testing this manually using the web UI, it looks like the remote address contains the same host portion each time but often (each request, even?) changes port numbers. This suggests that it might actually be more space efficient to split the `String remoteAddr` (`"127.0.0.1:1234"`) into a `String host` (`"127.0.0.1"`) and an `int port` (`1234`) fields. This way, the commonly re-appearing host strings can be interned effectively too. This would also make it easier for somebody analyzing the recording to sort/filter by requesting client, since the request host/IP is a distinct field.